x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/WriteActionsTests.java

```diff
@@ -101,64 +104,77 @@ public void testBulk() {
                 .add(new UpdateRequest("test4", "type", "id").doc(Requests.INDEX_CONTENT_TYPE, "field", "value"))
                 .add(new UpdateRequest("missing", "type", "id").doc(Requests.INDEX_CONTENT_TYPE, "field", "value")).get();
         assertTrue(bulkResponse.hasFailures());
-        assertEquals(13, bulkResponse.getItems().length);
-        assertFalse(bulkResponse.getItems()[0].isFailed());
-        assertEquals(DocWriteRequest.OpType.INDEX, bulkResponse.getItems()[0].getOpType());
-        assertEquals("test1", bulkResponse.getItems()[0].getIndex());
-        assertTrue(bulkResponse.getItems()[1].isFailed());
-        assertEquals(DocWriteRequest.OpType.INDEX, bulkResponse.getItems()[1].getOpType());
-        assertEquals("index1", bulkResponse.getItems()[1].getFailure().getIndex());
+        assertThat(bulkResponse.getItems().length, equalTo(13));
+        assertThat(bulkResponse.getItems()[0].getFailure(), nullValue());
```

Comment:
90% of the value of this change is here. Asserting `isFailed() == false` tells you very little if the item did in fact fail.
Asserting on the `failure` being `null` tells you _what went wrong_ when the assertion is not met.